apiserver/facades/agent/uniter/access.go

```diff
@@ -13,23 +13,21 @@ import (
 	"github.com/juju/juju/apiserver/facade"
 	coreapplication "github.com/juju/juju/core/application"
 	"github.com/juju/juju/core/unit"
-	"github.com/juju/juju/state"
 )
 
-func applicationAccessor(authorizer facade.Authorizer, st *state.State) common.GetAuthFunc {
-	return func(context.Context) (common.AuthFunc, error) {
+func applicationAccessor(authorizer facade.Authorizer) common.GetAuthFunc {
+	return func(ctx context.Context) (common.AuthFunc, error) {
 		switch tag := authorizer.GetAuthTag().(type) {
 		case names.ApplicationTag:
 			return func(applicationTag names.Tag) bool {
 				return tag == applicationTag
 			}, nil
 		case names.UnitTag:
-			entity, err := st.Unit(tag.Id())
+			unitName, err := unit.NewName(tag.Id())
```

Comment:
This is a behavior change, we're no longer verifying that the AuthTag entity exists. My question is, is that okay?

For the machine accessor, we're verifying the machine exists by virtue of getting the machine ID.

Reply:
Almost all accessors verify the tag kind, nothing else. Will find out soon enough if the unit or application does not exist.

Reply:
I can also call through to the application service to check if the unit exists to preserve behaviour exactly, should I do that?